controller/controller.js

```diff
@@ -1,16 +1,67 @@
-// c = controller, m = model, v = view, db = database
 const model = require("../model/model");
-
-// TODO: find out how error catching is meant to be, repeating .catch here seems not very DRY
+const errors = require("../controller/errors.js");
 
 // /api/topics Endpoints
-exports.getApiTopics = (request, response) => {
-    model.selectTopics().then((selectTopicsResponseObject) => {
+const getApiTopics = (request, response, next) => {
+    model.topics.selectTopics().then((selectTopicsResponseObject) => {
         response.status(200).send(selectTopicsResponseObject);
     }).catch((err) => {
-        console.log(err);
-    });
+        next(err);
+      });
+};
+
+// /api/articles Endpoints
+const getApiArticles = (request, response, next) => {
+    model.articles.selectArticles().then((responseObject) => {
+
+        response.status(200).send(responseObject);
+
+    }).catch((err) => {
+        next(err);
+      });
+};
+
+// /api/articles/:article_id Endpoints
+const getArticleById = (request, response, next) => {
+    const articleId = request.params["article_id"];
+
+    model.articles.selectArticleByArticleId(articleId).then((selectArticleByArticleIdResult) => {
+        const responseObject = {};
+        responseObject.article = selectArticleByArticleIdResult.rows[0];
+
+        response.status(200).send(responseObject);
+    }).catch((err) => {
+        next(err);
+      });
+
+};
+
+// 6. GET /api/articles/:article_id/comments
+const getCommentsByArticleId = (request, response, next) => {
+    const articleId = request.params["article_id"];
+
+    model.comments.selectCommentsByArticleId(articleId).then((selectCommentsByArticleIdOutput) => {
+        const feCommentRemoveArticleId = function (element, index) {
+            delete returnedCommentsArray[index]["article_id"];
+        };
```

Comment:
there should be no further data manipulation in the controller. If you don't want the article_id then exclude it from the SELECT in the model